staging/src/k8s.io/apimachinery/pkg/runtime/serializer/codec_factory.go

```diff
@@ -92,15 +108,34 @@ type CodecFactory struct {
 	legacySerializer runtime.Serializer
 }
 
+// CodecFactoryOptions holds the options for configuring CodecFactory behavior
+type CodecFactoryOptions struct {
+	// Strict: configures all serializers in strict mode
+	Strict bool
+	// Pretty: includes a pretty serializer along with the non-pretty one
+	Pretty bool
```

Comment:
should we add similar comment/details as in SerializerOptions, that Pretty is only for JSON encoding and Strict is only for decoding?

Reply:
I'm not sure.
Maintaining these docs is tougher if the behavior of `SerializerOptions` changes.

I'm fine to add more detail here -- it's good for library users.
Let me know what you think.

Reply:
I could add a note that refers to the SerializerOptions godoc?

Reply:
i'm leaning towards adding the notes, but let's see what the others think.

Reply:
Personally, I'd prefer the details here. I had to trace through the code to see why `CodecFactoryOptions` ignored the value for field `Pretty` for the yamlSerializer in `newSerializersForScheme`.